lib/log_ratelimit.c

```diff
@@ -49,24 +53,30 @@ log_ratelimit_reset(struct log_ratelimit_state *lrs, uint64_t now)
 	lrs->printed = 0;
 	if (lrs->suppressed > 0) {
 		rte_log(RTE_LOG_NOTICE, gatekeeper_logtype,
-			"GATEKEEPER: %u log entries were suppressed at lcore %u during the last ratelimit interval\n",
-			lrs->suppressed, rte_lcore_id());
+			"GATEKEEPER %s: %u log entries were suppressed at lcore %u during the last ratelimit interval\n",
+			lrs->block_name, lrs->suppressed, rte_lcore_id());
 	}
 	lrs->suppressed = 0;
 	lrs->end = now + lrs->interval_cycles;
 }
 
 void
-log_ratelimit_state_init(unsigned lcore_id, uint32_t interval, uint32_t burst)
+log_ratelimit_state_init(unsigned lcore_id, uint32_t interval, uint32_t burst,
+	uint32_t log_level, const char *block_name)
 {
 	struct log_ratelimit_state *lrs;
 
 	RTE_VERIFY(lcore_id < RTE_MAX_LCORE);
 
 	lrs = &log_ratelimit_states[lcore_id];
+
+	RTE_VERIFY(strlen(block_name + 1) < sizeof(lrs->block_name));
```

Comment:
BUG: `(block_name + 1)` is an undefined pointer here. Also, if you do want to keep the `+ 1` outside of `strlen()`, the inequality must be `<=`. I recommend to just dropping the `+ 1` to fix everything.